onnxruntime/core/providers/cuda/cuda_call.cc

```diff
@@ -94,7 +94,7 @@ bool CudaCall(ERRTYPE retCode, const char* exprString, const char* libName, ERRT
       size_t hostname_len = 0;
       char* hostname = nullptr;
       if (-1 == _dupenv_s(&hostname, &hostname_len, "COMPUTERNAME"))
-        hostname = "?";
+        hostname = const_cast<char*>("?");
```

Comment:
can we avoid const_cast?

Reply:
I would prefer to leave it as a TODO as it is not related to C++ 14/17 upgrade.  Currently this PR is already not small.

Reply:
ok. TODO comment?